src/scripts/glue_jobs/glue_job.py

```diff
@@ -73,30 +74,32 @@ def execute(self):
 
     # Resolve the data type mismatch by comparing it with table schema and store the processed data back to s3
     def resolve_choice_and_write_output_frame(self, transformed_frame, partition_key):
-        job_name = self.args["JOB_NAME"]
-        if transformed_frame.stageErrorsCount() > 0:
-            self.log_errors(transformed_frame)
+        self.check_and_log_errors(transformed_frame)
 
         output_frame = transformed_frame.resolveChoice(choice='match_catalog', database=self.args['DATABASE_NAME'],
                                                        table_name=self.args['TABLE_NAME'])
-        self.logger.info("Total output records write to s3 for job {} is {}".format(job_name,
-                                                                                    str(output_frame.count())))
+        self.check_and_log_errors(output_frame)
+
         if output_frame.count() > 0:
+            self.logger.info("The total number of records saved to s3 is {}".format(output_frame.count()))
```

Comment:
We are doing a count twice which can be very slow, I would leave this out unless we need it